### PR TITLE
Provide url

### PR DESCRIPTION
**Pull request by robertberry**
This is so we can check whether IDs queries are going to be rejected due to their being larger than the maximum URL size

CC @maxharlow @adamnfish

### CODE REVIEW

File: src/main/scala/com.gu.contentapi.client/Limits.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.gu.contentapi.client
+
+object Limits {
+  val UrlSize = 2048
```
Comment (adamnfish) [inline src/main/scala/com.gu.contentapi.client/Limits.scala]: I thought you said the limit was double that?

Comment (robertberry) [inline src/main/scala/com.gu.contentapi.client/Limits.scala]: Oops yes you're right 4096

**Comment by mchv** [comment]
Please note the`4096` limit is not exact from the client side as the host is used is the computation and will differ from `content.guardianapis.com` on the server side.

**Comment by robertberry** [comment]
Oh right ... Hmm, so is the host the EC2 host?

I basically need some kind of lower bound.

**Comment by robertberry** [comment]
@mchv I've updated the calculation for max url size, can you give it a once over please?

**Comment by mchv** [comment]
:+1:  we need to update the changelog (with as well the changes from https://github.com/guardian/content-api-scala-client/pull/59) and bump the version to `4.0` and release a new version.

**Comment by robertberry** [comment]
Ok cool, I'll do that and make a release now

**Comment by philwills** [comment]
This is lovely, I was just coming to try and do something like this myself and find the lovely Mr. Berry has beat me to  it.

Can't help thinking it would be nice to have a prettier toString on the Query objects though.

**Comment by adamnfish** [comment]
Since query objects are now pure, they don't have access to all the stuff though (e.g. API key, host). They're only able to get hold of the parameters that have been configured on that object. I think this approach is generally going to be more useful.

**Comment by robertberry** [comment]
@philwills Could do that, it wouldn't have the host or API key, though (as they belong to the client itself).

**Comment by robertberry** [comment]
Aha, Adam beat me to this.

BTW @adamnfish, given that they're now pure, it seems weird to have the constructors as methods on the client.

**Comment by adamnfish** [comment]
Yes, that's a leftover helper method from the old approach. We could remove that as well although in MAPI it's actually quite useful because there are roughly 10^6 things called ItemQuery and client.item is unambiguous and terse...

**Comment by robertberry** [comment]
Hahaha, which idiot did that? :eyes:

**Comment by philwills** [comment]
Yea, I'm quite happy with some partial string that's not a well-formed URL, I just want something more meaningful than SearchQuery#fiejowf90f09209i02

**Comment by robertberry** [comment]
Good call ... I'll do that, one sec.

**Pull request merged**